src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala

```diff
@@ -162,8 +167,8 @@ class CassandraSnapshotStore(cfg: Config) extends SnapshotStore with CassandraSt
   }
 
   override def deleteAsync(persistenceId: String, criteria: SnapshotSelectionCriteria): Future[Unit] = {
-    preparedSelectSnapshotMetadataForDelete.flatMap { prepStmt =>
-      metadata(prepStmt, persistenceId, criteria).flatMap { mds =>
+    preparedSelectSnapshotMetadata.flatMap { prepStmt =>
+      metadata(prepStmt, persistenceId, criteria, limit = -1).flatMap { mds =>
```

Comment:
Yuck, another magic number! Can we do it with `Option` instead, or is this to save allocs?